canonical-webteam/deployment-configs
# Add new redirects and config fixes

Add new redirects:

- www.netplan.io -> netplan.io
- ubuntu-china.cn -> cn.ubuntu.com
- staging.ubuntu-china.cn -> cn.staging.ubuntu.com
- canonical.com -> www.canonical.com
- staging.canonical.com -> www.staging.canonical.com

Additionally:

- Add `?` to the end of all Nginx `rewrite` rules, to avoid the [double query parameter problem](https://stackoverflow.com/questions/28081159/nginx-request-uri-has-duplicated-query-parameter)
- Update `ingress.kubernetes.io` to the new `nginx.ingress.kubernetes.io` everywhere
- Add `path: /` to snapcraft.io ingress to make it more similar to other ingress configs
- Remove `dev.ubuntu.com` and `developers.ubuntu.com` from the `tls:` config blog, as the TLS cert doesn't match those domains

QA
--

``` bash
$ minikube start
$ kubectl config use-context minikube
$ cd ingresses
$ cat production/netplan.io.yaml production/cn.ubuntu.com.yaml production/www.canonical.com.yaml staging/cn.staging.ubuntu.com.yaml staging/www.staging.canonical.com.yaml | sed '/namespace:/d' | kubectl apply -f -
$ curl -sI -H 'Host: www.netplan.io' `minikube ip`/?fish=chips | grep -i location
Location: https://netplan.io/?fish=chips
$ curl -sI -H 'Host: ubuntu-china.cn' `minikube ip`/?fish=chips | grep -i location
Location: https://cn.ubuntu.com/?fish=chips
$ curl -sI -H 'Host: canonical.com' `minikube ip`/?fish=chips | grep -i location
Location: https://www.canonical.com/?fish=chips
$ curl -sI -H 'Host: staging.ubuntu-china.cn' `minikube ip`/?fish=chips | grep -i location
Location: https://cn.staging.ubuntu.com/?fish=chips
$ curl -sI -H 'Host: staging.canonical.com' `minikube ip`/?fish=chips | grep -i location
Location: https://www.staging.canonical.com/?fish=chips
```

Check all redirects work as expected ^.
